include/envoy/filesystem/filesystem.h

```diff
@@ -112,12 +112,29 @@ class Instance {
   virtual std::string fileReadToEnd(const std::string& path) PURE;
 
   /**
-   * @path file path to split
+   * @param path file path to split
    * @return PathSplitResult containing the parent directory of the input path and the file name
    * @note will throw an exception if path does not contain any path separator character
    */
   virtual PathSplitResult splitPathFromFilename(absl::string_view path) PURE;
 
+  /**
+   * @param path1 first path component.
+   * @param path2 second path component.
+   * @return composite path with the second path component joined to the first.
+   */
+  virtual std::string joinPath(absl::string_view path1, absl::string_view path2) PURE;
+
+  /**
+   * Resolve a symlink path.
+   * @param path to resolve.
+   * @return resolved path. If there is an error, an exception is thrown, unless
+   *         the error is a result of path not being a symlink, in which case
+   *         the supplied path is returned.
+   * @throw EnvoyException if the path can't be resolved.
+   */
+  virtual std::string readSymlink(absl::string_view path) PURE;
```

Comment:
@wrowe would it be reasonable to land this on Linux/Mac and tag as unsupported on Windows, leaving a tracking issue for @envoyproxy/windows-dev? I think for new platform level APIs it would be good to have the Windows experts own these.

Reply:
Also, see the other comments I tagged you on, we might want this to be full blown path canonicalization, similar to the APR example you shared.

Reply:
Not needed any more since I switched to hash-based atomic reads.